lighthouse-core/scripts/lantern/collect/collect.js

```diff
@@ -0,0 +1,316 @@
+    archive.directory(archiveDir, false);
+    archive.finalize();
+  });
+}
+
+const log = new ProgressLogger();
+
+/** @type {UrlResults[]} */
+const summary = loadSummary();
+
+/**
+ * Resume state from previous invocation of script.
+ * @return {UrlResults[]}
+ */
+function loadSummary() {
+  if (fs.existsSync(summaryPath)) {
+    /** @type {UrlResults[]} */
+    const summary = JSON.parse(fs.readFileSync(summaryPath, 'utf-8'));
+    // Remove data if no longer in URLS.
+    return summary.filter(urlSet => URLS.includes(urlSet.url));
+  } else {
+    return [];
+  }
+}
+
+function saveSummary() {
+  fs.writeFileSync(summaryPath, JSON.stringify(summary, null, 2));
+}
+
+/**
+ * @param {string} url
+ * @return {Promise<string>}
+ */
+function fetch(url) {
+  return new Promise((resolve, reject) => {
+    https.get(url, (res) => {
+      let data = '';
+      res.on('data', (chunk) => data += chunk);
+      res.on('end', () => resolve(data));
+      res.on('error', (err) => reject(err));
+    });
+  });
+}
+
+/**
+ * @param {string} url
+ */
+async function startWptTest(url) {
+  const apiUrl = new URL('https://www.webpagetest.org/runtest.php');
+  apiUrl.searchParams.set('k', WPT_KEY);
+  apiUrl.searchParams.set('f', 'json');
+  // Keep the location constant. Use Chrome and 3G network conditions.
+  apiUrl.searchParams.set('location', 'Dulles:Chrome.3G');
+  apiUrl.searchParams.set('lighthouse', '1');
+  // Makes the trace file available over /getgzip.php.
+  apiUrl.searchParams.set('lighthouseTrace', '1');
+  // Disables some things that WPT does, such as a "repeat view" analysis.
+  apiUrl.searchParams.set('type', 'lighthouse');
+  apiUrl.searchParams.set('mobile', '1');
+  // apiUrl.searchParams.set('mobileDevice', '1');
+  apiUrl.searchParams.set('url', url);
+  const wptResponseJson = await fetch(apiUrl.href);
+  const wptResponse = JSON.parse(wptResponseJson);
+  if (wptResponse.statusCode !== 200) {
+    throw new Error(`unexpected status code ${wptResponse.statusCode} ${wptResponse.statusText}`);
+  }
+
+  return {
+    testId: wptResponse.data.testId,
+    jsonUrl: wptResponse.data.jsonUrl,
+  };
+}
+
+/**
+ * @param {string} url
+ * @return {Promise<Result>}
+ */
+async function runForUnthrottled(url) {
+  const artifactsFolder = `${LH_ROOT}/.tmp/collect-traces-artifacts`;
+  await new Promise((resolve, reject) => {
+    execFile('node', [
+      path.join(LH_ROOT, 'lighthouse-cli'),
+      url,
+      `-G=${artifactsFolder}`,
+    ], (_, stderr) => {
+      if (stderr) reject(stderr);
+      else resolve();
+    });
+  });
+  const trace = fs.readFileSync(`${artifactsFolder}/defaultPass.trace.json`, 'utf-8');
+  return {
+    trace,
+  };
+}
+
+/**
+ * @param {string} url
+ * @return {Promise<Result>}
+ */
+async function runForMobile(url) {
+  const {testId, jsonUrl} = await startWptTest(url);
+  if (DEBUG) log.log({testId, jsonUrl});
+
+  // Poll for the results every x seconds, where x = position in queue.
```

Comment:
seems a bit aggressive? (especially running nine at a time)

Reply:
It's an underestimate, actually. It takes longer than 1 second per position in queue.

Reply:
I think that's exactly his point @connorjclark , it's a bit aggressive to the API to do this so frequently :)

Reply:
I see. 1.5 ?